### PR TITLE
OPENEUROPA-1843: Paragraph variant doesn't stay selected when resaving the content

### PR DESCRIPTION
**Pull request by brummbar**
### Description

Paragraph variant doesn't stay selected when resaving the content.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
         "drupal/drupal-extension": "~4.0",
         "drush/drush": "~9.0",
         "nikic/php-parser": "~3.0",
+        "openeuropa/behat-transformation-context" : "~0.1",
         "openeuropa/code-review": "~1.0@beta",
         "openeuropa/drupal-core-require-dev": "^8.6",
         "openeuropa/task-runner": "~1.0@beta"
```
Comment (ademarco) [inline composer.json]: To solve lowest dependencies build failure use here `~1.0-beta5`.
```
        "openeuropa/task-runner": "~1.0-beta5"
```

File: tests/Behat/ParagraphsContext.php [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+      throw new \Exception(sprintf('No option is selected for the "%s" select.', $select));
+    }
+
+    if ($option_element->getText() !== $option) {
+      throw new \Exception(sprintf('The option "%s" was not selected, "%s" was selected.', $option, $option_element->getHtml()));
+    }
+  }
+
+  /**
+   * Returns a collection of fields that belong directly to a paragraph.
```
Comment (upchuk) [inline tests/Behat/ParagraphsContext.php]: I would say this is: 
```
Finds and returns the paragraphs at a given position.
```

File: tests/features/variants.feature [hunk 3]
```diff
@@ -91,3 +92,45 @@ Feature: Paragraph types show different fields based on variants.
     When I select "Date" from "Variant"
     And I wait for AJAX to finish
     Then the following fields should be present "Layout, Variant, Link, Title, Description, Meta, URL, Link text, Day, Month, Year"
+
+  # This scenario covers the bug discovered in OPENEUROPA-1843.
+  @javascript @run
```
Comment (upchuk) [inline tests/features/variants.feature]: Remove the `@run` ~~Danny~~ Fra.

File: tests/features/variants.feature [hunk 4]
```diff
@@ -91,3 +92,45 @@ Feature: Paragraph types show different fields based on variants.
...
+    Given I am logged in as a user with the "Editor" role
+    And I go to "the content management page"
+    And I click "Add content"
+    And I fill in "Title" with "Test page"
+    And I press "List additional actions"
+    And I press "Add Listing item block"
+    And I wait for AJAX to finish
+    And I select "One column" from "Layout" in the 1st "Listing item block" paragraph
+    And I fill in "Title" with "Block title" in the 1st "Listing item block" paragraph
+    And I select "Highlight" from "Variant" in the 1st "Listing item"
```
Comment (upchuk) [inline tests/features/variants.feature]: Any reasons for which you don't use the `paragraph` ending to each of these steps?

**Pull request merged**